.github/workflows/template-deploy.yml

```diff
@@ -0,0 +1,71 @@
+name: Deployment template
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+        description: "The environment to deploy to"
+        default: "dev"
+      subscriptionId:
+        type: string
+        required: true
+        description: "The Azure subscription ID to deploy to"
+        default: "analogio-dev"
+      targetBranch:
```

Comment:
Maybe we could just read the branch name of the Github env variable instead of having it as string input?